Controllers/Frontend/Adyen.php

```diff
@@ -103,20 +103,27 @@ public function resetValidPaymentSessionAction()
      */
     public function ajaxThreeDsAction()
     {
-        $this->Request()->setHeader('Content-Type', 'application/json');
-        $this->Front()->Plugins()->ViewRenderer()->setNoRender();
-
-        $postData = (array) $this->Request()->getPost();
-        $payload = array_intersect_key($postData, ['details' => true]);
-        $threeDSResult = (string) ($payload['details']['threeDSResult'] ?? '');
+        $threeDSResult = (string) ($this->Request()->getPost()['details']['threeDSResult'] ?? '');
         if ('' === $threeDSResult) {
-            $this->logger->error('3DS2 missing data', [
+            $this->logger->error('3DS missing data', [
                 'action' => $postData['action'] ?? [],
                 'threeDSResult' => substr($threeDSResult, -5),
                 'paymentData' => substr($postData['paymentData'] ?? '', -5),
             ]);
         }
 
+        $this->paymentDetailsAction();
+    }
+
+    /**
+     * @throws AdyenException
+     */
+    public function paymentDetailsAction()
```

Comment:
@acampos1916 
The 2 methods are split up, handlePaymentDetails is the main action,
the ThreeDsAction is to add some logging (the extra logging is useful during upgrading web components and debugging of 3DS actions)